datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -1488,9 +1489,59 @@ def load_config(yaml_str: str) -> Any:
         if process.returncode != 0:
             raise ValueError(f'Failed to load config: {stderr.decode()}')
 
-        decoded = stdout.strip().decode()
+        decoded_string = stdout.strip().decode()
+
+        # We need to process inf and nan in a special way since they are not valid Python literals.
+        # We'll replace them with placeholders to avoid parsing errors, and then restore them after parsing.
+
+        INF_PLACEHOLDER = '__PYTHON_INF__'
+        NEG_INF_PLACEHOLDER = '__PYTHON_NEG_INF__'
+        NAN_PLACEHOLDER = '__PYTHON_NAN__'
+
+        # AST Transformer to replace inf/nan names with placeholder constants
+        class _SpecialFloatValuesTransformer(ast.NodeTransformer):
+            def visit_Name(self, node: ast.Name) -> ast.AST:
+                if node.id == 'inf':
+                    return ast.Constant(value=INF_PLACEHOLDER)
+                elif node.id == 'nan':
+                    return ast.Constant(value=NAN_PLACEHOLDER)
+                return node
+
+            def visit_UnaryOp(self, node: ast.UnaryOp) -> ast.AST:
+                if isinstance(node.op, ast.USub) and isinstance(node.operand, ast.Name) and node.operand.id == 'inf':
+                    return ast.Constant(value=NEG_INF_PLACEHOLDER)
+                return self.generic_visit(node)
+
+        # Helper to restore placeholder strings to actual float values
+        def _restore_special_floats(data: Any) -> Any:
+            if isinstance(data, dict):
+                return {key: _restore_special_floats(value) for key, value in data.items()}
+            elif isinstance(data, list):
+                return [_restore_special_floats(item) for item in data]
+            elif isinstance(data, str):
+                if data == INF_PLACEHOLDER:
+                    return float('inf')
+                elif data == NEG_INF_PLACEHOLDER:
+                    return float('-inf')
+                elif data == NAN_PLACEHOLDER:
+                    return float('nan')
+            return data
```

Comment:
Thanks! This is actually pretty clever, thanks a lot! Could we extract the logic to its own module, just to avoid increasing the working memory of whoever reads this later 😅 .

Maybe having it within an `_ast_config.py` or something like this under `datadog_checks.base.checks` would work and then import it when we call the `load_config ` method? Expose a parse method that does everything and returns the final python dictionary representing the config.

Could we also add a comment on each `visit_*` method just to explain what each visit does? Not sure how familiar everyone is with AST, explaining that the Name visit receives names like varaibles or literals and UnaryOp receives unary operators like `-`, `+` or `~` maybe is enough.